private/source.rkt

```diff
@@ -82,20 +93,33 @@
     (define stx (source-read-syntax code))
     (define current-expand-observe (dynamic-require ''#%expobs 'current-expand-observe))
     (define visits-by-location (make-hash))
-    (define/guard (observe-event! sig val)
-      (guard (and (equal? sig 'visit) (syntax? val) (syntax-original? val)) else
-        (void))
-      (define loc (syntax-source-location val))
-      (guard (hash-has-key? visits-by-location loc) then
-        (void))
-      (hash-set! visits-by-location loc val))
+    (define others-by-location (make-hash))
+    (define/guard (add-original-location! hsh stx)
+      (guard (and (syntax? stx) (syntax-original? stx)) else (void))
+      (define loc (syntax-source-location stx))
+      (guard (hash-has-key? hsh loc) then (void))
+      (hash-set! hsh loc stx))
+    (define/match (observe-event! sig val)
+      [('visit val)
+       (add-original-location! visits-by-location val)]
+      [('letX-renames (list-rest trans-idss _ val-idss _))
```

Comment:
So, what _is_ a `'letX-renames` event? If you reverse engineered it by looking at some other code somewhere, could you add a comment with a link to that code so I can figure it out too?